### PR TITLE
BREAKING CHANGE: Grouping errors

### PR DESCRIPTION
**Pull request by galmail**
### What does this PR do?

It groups the errors under one Error object to simplify usage.

### Related issues

https://github.com/nearform/graphql-hooks/issues/302

### CODE REVIEW

**Comment by galmail** [comment]
Please note: the package version hasn't been changed. We probably need to bump the major version as these changes are not backward compatible.

File: .circleci/config.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ jobs:
     steps:
       - checkout
 
-      - restore_cache:
-          key: dependency-cache-{{ checksum "package.json" }}
+      # - restore_cache:
+      #     key: dependency-cache-{{ checksum "package.json" }}
```
[the remark this answers is not in the document]
Comment (galmail) [reply .circleci/config.yml]: yeah makes sense, I've changed it. Have a look and let me know what you think... in regards to backwards compatibility, if we do that we will end up with an error object which already contains fetchError, httpError and graphQLErrors, plus those again those same objects.. so not so nice...

**Comment by galmail** [comment]
thanks @bmullan91 I've also updated the way we generating the cache key to guarantee we always get a new key if one of the package.json files has changed.  @Joezo please have a look and let me know, thanks!

**Comment by Joezo** [comment]
@galmail 	I've rerun the build a few times and it never seems to restore the cache. It looks like it's generating a different checksum each time, could you take another look? I'm happy with this otherwise :)

**Comment by galmail** [comment]
@Joezo you're right! CircleCI actually generates a new checksum whenever it creates a new file.. so I've removed the script I've created and concatenated all the checksums and now it is working.

**Comment by bmullan91** [comment]
@all-contributors please add @galmail for code bug doc test

**Pull request merged**